src/0install-solver/sat.ml

```diff
@@ -200,17 +199,13 @@ module Make (User : USER) = struct
   ;;
 
   (* For nicer log_debug messages *)
-  let name_lit f (sign, var) =
+  let name_lit (sign, var) =
     match sign with
-    | Pos -> User.pp f var.obj
-    | Neg -> Format.fprintf f "not(%a)" User.pp var.obj
+    | Pos -> User.pp var.obj
+    | Neg -> Pp.text "not(" ++ User.pp var.obj ++ Pp.char ')'
```

Comment:
textf

Reply:
I don't see how one would use `Pp.textf` here. If we keep `User.pp` as `User.t -> _ Pp.t`, then the `%a` conversion does not work.
If we wish to use it as a string that's possible, but we loop back to removing a lot of Pp.t everywhere

Reply:
Maybe my regexp game isn't strong enough, but I can't find a single use of `Pp.textf "...%a"` in the entire dune codebase or [anywhere else in the OCaml world](https://sherlocode.com/?q=Pp.textf%20%22%5C.%5C*%25a), even though it is technically possible, it just requires a very weirdly typed conversion function (`unit -> <your type, here User.t> -> string`)

Reply:
OK never mind about this then, there are better things to focus on.